### ISSUE TITLE
tracee GOB format missing detections

### ISSUE THREAD

**Issue by AsafEitani** [aquasecurity/tracee#1476]
When running tracee-ebpf and rules with GOB output, detections get missed sometimes.
It is unknown if the problem reside in the ebpf part or the rules but there are no reported lost events.
Running the same behavior several times yields different results.
The issue occurs both when piping tracee-ebpf output to tracee-rules and when the output is saved to a file and then read by tracee-rules.

The issue seems to be limited to GOB and doesn't occur in JSON output.

I'm still investigating and will update the status here.

**Comment by rafaeldtinoco** [comment]
Check first if this isn't related to:

https://github.com/aquasecurity/tracee/issues/1456 and mitigated by https://github.com/aquasecurity/tracee/pull/1470.

Could you give it a try ? You can compile the code from PR #1470 and execute it by using:

```
sudo ./dist/tracee-ebpf -o format:json -o option:parse-arguments -o option:cache-events
```

It will cache up to 1M events and consume up to ~1.1GB of ram. It caches both, JSON and GOB, so .. in both cases detection should happen. If, with GOB, it doesn't happen, then it will confirm your suspicious (but I would test first this branch to make sure you're not observing other issue, already known).

**Comment by AsafEitani** [comment]
So I found out it is an old problem we had where the container id is empty for some of the events.
This was fixed but I was testing on v0.6.5 so we I will test again on main and will update.
